AutorestSwiftTest/AutoRestResourceFlatteningTest.swift

```diff
@@ -44,21 +44,304 @@ class AutoRestResourceFlatteningTest: XCTestCase {
         )
     }
 
-//    func test_Head_success200() throws {
-//        let expectation = XCTestExpectation(description: "Call head200 succeed")
-//        let failedExpectation = XCTestExpectation(description: "Call head200 failed")
-//        failedExpectation.isInverted = true
-//
-//        client.head200() { result, _  in
-//            switch result {
-//                case .success:
-//                 expectation.fulfill()
-//               case let .failure(error):
-//                print("test failed. error=\(error.message)")
-//                failedExpectation.fulfill()
-//            }
-//        }
-//
-//        wait(for: [expectation], timeout: 5.0)
-//    }
+    private func errorDetails(for error: AzureError, withResponse response: HTTPResponse?) -> String {
+        var details: String
+        if let data = response?.data {
+            details = String(data: data, encoding: .utf8)!
+        } else {
+            details = error.message
+        }
+        return details
+    }
+
+    func test_resourceFlattening_getArray() throws {
+        let expectation = XCTestExpectation(description: "Call autorestresourceflatteningtestservice.getArray succeed")
+
+        client.autorestresourceflatteningtestservice.getArray { result, httpResponse in
+            switch result {
+            case .success:
+                expectation.fulfill()
+            case let .failure(error):
+                let details = self.errorDetails(for: error, withResponse: httpResponse)
+                print("test failed. error=\(details)")
+                XCTFail("Call autorestresourceflatteningtestservice.getArray failed")
+            }
+        }
+        wait(for: [expectation], timeout: 5.0)
+    }
+
+    func test_resourceFlattening_putArray() throws {
+        let expectation =
+            XCTestExpectation(
+                description: "Call autorestresourceflatteningtestservice.putArray succeed"
+            )
+        let array = [
+            Resource(id: nil, type: nil, tags: ["tag1": "value1", "tag2": "value3"], location: "West US", name: nil),
+            Resource(id: nil, type: nil, tags: nil, location: "Building 44", name: nil)
```

Comment:
@sacheu it seems like this initializer should have `= nil` defaults for its optional arguments to simplify. Do you agree?

Reply:
I have fixed this.